WordPress/gutenberg
# [RNMobile] Add radial gradient infrastructure

## Description

That PR is adding infrastructure for gradients to switch between both types along with changing a linear gradient angle.

That PR is based on i[nitial radial gradient PR](https://github.com/WordPress/gutenberg/pull/22384)

Ref to gb-mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2277
Testing ios: https://github.com/wordpress-mobile/WordPress-iOS/pull/14154
Testing android: https://github.com/wordpress-mobile/WordPress-Android/pull/11999


## How has this been tested?

1. Open mobile app
2. Add block with color support (Button / Cover)
3. Open block settings
4. Press `gradient` segment

Expect:
* `Customize Gradient` control appeared

5. Press `Customize Gradient`
6. Change `Angle`

Expect: 
* Gradient angle is changing

7. Press `Radial` type

Expect: 
* Gradient changed to radial

8. Press `back` button

Expect:
* Color palette list is scrolled to the end
* `Custom` text along with selected and customized gradient appeared

9. Press `solid` segment
10. Press `gradient` segment

Expect:
* Color palette is scrolling to the end (to the custom gradient)

11. Press `Custom` or selected color swatch

Expect:
* `Customize Gradient` subsheet is opened


## Screenshots 

* ios

<details>

Color Settings | Customize Gradient
--- | ---
<img width="484" alt="Screenshot 2020-05-22 at 14 43 08" src="https://user-images.githubusercontent.com/22746080/82669310-87965000-9c3b-11ea-9184-a83f53f924c4.png"> | <img width="484" alt="Screenshot 2020-05-22 at 14 43 11" src="https://user-images.githubusercontent.com/22746080/82669312-88c77d00-9c3b-11ea-8925-d17b1ae54842.png">

</details>


* android 

<details>

Color Settings | Customize Gradient
--- | ---
<img width="427" alt="Screenshot 2020-05-22 at 14 42 50" src="https://user-images.githubusercontent.com/22746080/82669413-aeed1d00-9c3b-11ea-9423-26ddf84e2132.png"> | <img width="427" alt="Screenshot 2020-05-22 at 14 42 54" src="https://user-images.githubusercontent.com/22746080/82669417-b01e4a00-9c3b-11ea-806c-15abed4305b9.png">


</details>

## Types of changes

Adding new control in color settings called `Customize Gradient` which moves into subsheet with controls to change gradient type and linear gradient angle.

## Checklist:
